policy-controller/src/admission.rs

```diff
@@ -447,6 +439,10 @@ impl Validate<HttpRouteSpec> for Admission {
     }
 }
 
+// NOTE: This implementation is currently unused, as the policy controller only
+// supports indexing the `policy.linkerd.io` version of the `HTTPRoute` CRD.
+// This implementation will be used if support for the
+// `gateway.networking.k8s.io` CRD is added again in the future.
```

Comment:
similarly, if it's actually unused it seems fine to remove it.

Reply:
removing the `Validate` impl maeks some of the functions it depends on dead code, i'm going to remove them as well, unless you'd prefer to keep them with `#[allow(dead_code)]` added?